### ISSUE TITLE
1.6.0 not working at my MMDVM_HS_DUAL

### ISSUE THREAD

**Issue by dl1bz** [juribeparada/MMDVM_HS#145]
I have a MMDVM_HS_DUAL with 12,880 MHz. With FW 1.5.2 all worked, if I compile the newest 1.6.0 MMDVMHost says NAK from modem. MMDVMCal also not working. Flashing backward to 1.5.2 bring all back, the modem works correct.

73 Heiko, DL1BZ

**Comment by KE7FNS** [comment]
> ok, here is my solution: a) MMDVMCal don't work I delete in an actual MMDVMCal all serial speeds higher than 115200. That brings me MMDVMCal back to work with the MMDVM_HS_DUAL and 1.6.0. I think we need a command line parameter with defined SerialSpeed in MMDVMCal ?!? You cannot put a defined serial speed if you start MMDVMCal.
> 
> b) MMDVMHost don't start I need to set UARTSpeed=115200 in MMDVM.ini With UARTSpeed=460800 MMDVMHost don't start with this MMDVM_HS_DUAL.
> 
> For now all is working now. But I think there are a few things need to be fixed with the newer, higher serial speed parameters.

The serial speed for MMDVM_HS is locked at 115200, this hasn't been changed to any higher speeds.  The higher speeds were only added for external FM compatibility and since MMDVM_HS  doesn't currently support FM, the higher speeds are not actually needed.

https://github.com/juribeparada/MMDVM_HS/blob/d4cb546a7c625a4c9949b949baf6a6e246884180/SerialPort.cpp#L589

**Comment by dl1bz** [comment]
Yes, but the *lastest* MMDVMCal don't detect the modem anymore. There is no option (at command line) to bring the serial speed down. Only if I modify the source code of MMDVMCal and delete higher serial speeds as 115200 the MMDVMCal come back to work with my MMDVM_HS_DUAL and 1.6.0. Earlier there was not such a problem. That was what I meant with "a few things need to be fixed" :)
The FW 1.6.0 itself works now.

73 Heiko, DL1BZ

**Comment by g4klx** [comment]
Thanks for pointing that out. I’ll try and fix it this evening UK time.


Sent from Yahoo Mail for iPhone


On Monday, November 1, 2021, 10:31, Heiko ***@***.***> wrote:




Yes, but the lastest MMDVMCal don't detect the modem anymore. There is no option (at command line) to bring the serial speed down. Only if I modify the source code of MMDVMCal and delete higher serial speeds as 115200 the MMDVMCal come back to work with my MMDVM_HS_DUAL and 1.6.0. Earlier there was not such a problem. That was what I meant with "a few things need to be fixed" :)
The FW 1.6.0 itself works now.

73 Heiko, DL1BZ

**Comment by dl1bz** [comment]
> Thanks for pointing that out. I’ll try and fix it this evening UK time. Sent from Yahoo Mail for iPhone On Monday, 
Great Jonathan. If you fix it I will try again. Thanks a lot.

73 Heiko, DL1BZ

**Comment by KE7FNS** [comment]
> Yes, but the _lastest_ MMDVMCal don't detect the modem anymore. There is no option (at command line) to bring the serial speed down. Only if I modify the source code of MMDVMCal and delete higher serial speeds as 115200 the MMDVMCal come back to work with my MMDVM_HS_DUAL and 1.6.0. Earlier there was not such a problem. That was what I meant with "a few things need to be fixed" :) The FW 1.6.0 itself works now.

Problems with MMDVMCal belong in the MMDVMCal repo.

> Thanks for pointing that out. I’ll try and fix it this evening UK time

@g4klx 

I've got a patch solution for MMDVMCal on my repo, but I'm my opinion it is still suboptimal, even though it works correctly in my testing.  Personally I think it would be best for MMDVMCal/MMDVMHost to try all valid serial port speeds to "autodetect" a modem, or at least try 115200, 230400, and 460800, since those are the ones you now define in the MMDVM config.h  That way all compatibility with older firmware for MMDVM and MMDVM_HS is still maintained.

https://github.com/KE7FNS/MMDVMCal/tree/KE7FNS-patch-1

**Comment by g4klx** [comment]
I dislike auto-baud intensely, sorry.
I've updated MMDVMCal to take a non-optional speed parameter, which should fix things for people. 115200 for hotspots, and 460800 for modern MMDVM modem versions. It also allows for 230400 baud.

    On Monday, 1 November 2021, 18:47:16 GMT, Jason Kraemer ***@***.***> wrote:  
 
 



Yes, but the lastest MMDVMCal don't detect the modem anymore. There is no option (at command line) to bring the serial speed down. Only if I modify the source code of MMDVMCal and delete higher serial speeds as 115200 the MMDVMCal come back to work with my MMDVM_HS_DUAL and 1.6.0. Earlier there was not such a problem. That was what I meant with "a few things need to be fixed" :) The FW 1.6.0 itself works now.


Problems with MMDVMCal belong in the MMDVMCal repo.


Thanks for pointing that out. I’ll try and fix it this evening UK time


@g4klx

I've got a patch solution for MMDVMCal on my repo, but I'm my opinion it is still suboptimal, even though it works correctly in my testing. Personally I think it would be best for MMDVMCal/MMDVMHost to try all valid serial port speeds to "autodetect" a modem, or at least try 115200, 230400, and 460800, since those are the ones you now define in the MMDVM config.h That way all compatibility with older firmware for MMDVM and MMDVM_HS is still maintained.

https://github.com/KE7FNS/MMDVMCal/tree/KE7FNS-patch-1

**Comment by AD5IA** [comment]
I checked the 1.6.0 MMDVM_HS source code. the USART1 speed is set to 115200, and the Pi-start also set to 115200. After flash the 1.6.0 to STM32, it turn out works with pi-star 4.1.6, but fail to work with 4.1.5 and 3.4.17.  Is there any other reason? how to make it work with 4.1.5 and 3.4.17? Thanks for any help.
